packages/typography/src/utility/StaticWidthText.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import Box from '@leafygreen-ui/box';
+import { getNodeTextContent, HTMLElementProps } from '@leafygreen-ui/lib';
+import { css, cx } from '@leafygreen-ui/emotion';
+
+type StaticWidthTextProps<T extends keyof JSX.IntrinsicElements> = Omit<
+  HTMLElementProps<T>,
+  'children'
+> & {
+  /**
+   * Text to render
+   */
+  children: string;
+  /**
+   * The maximum future weight of the text. Determines the width of the component.
+   * @default 700
+   */
+  maxFontWeight?: React.CSSProperties['fontWeight'];
+  /**
+   * Defines the pseudo element used to force the element width
+   */
+  pseudoElement?: 'before' | 'after';
+  as?: T;
+};
+
+const staticWidthTextStyle = ({
+  pseudoElement,
+  maxFontWeight,
+}: Pick<StaticWidthTextProps<any>, 'pseudoElement' | 'maxFontWeight'>) => {
+  const pseudoSelector = `&:${pseudoElement}`;
+
+  return css`
+    /* Inherit relevant properties from the parent */
+    font-weight: inherit;
+    font-size: inherit;
+
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
```

Comment:
I think you should also add in the comments that this component adds an ellipsis when text overflows